inference-engine/src/mkldnn_plugin/emitters/cpu_generator.cpp

```diff
@@ -0,0 +1,191 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets/pass/assign_registers.hpp"
+#include "snippets/pass/vector_to_scalar.hpp"
+#include "snippets/pass/insert_load_store.hpp"
+#include "snippets/register_info.hpp"
+#include "snippets/snippets_isa.hpp"
+
+#include <ngraph/pass/visualize_tree.hpp>
+#include <ngraph/rt_info.hpp>
+#include <ngraph/pass/manager.hpp>
+
+#include <string>
+#include <iostream>
+#include <array>
+
+#include "cpu_generator.hpp"
+#include "jit_snippets_emitters.hpp"
+#include "jit_eltwise_emitters.hpp"
+#include "jit_mkldnn_emitters.hpp"
+#include "jit_mkldnn_ext_emitters.hpp"
+
+using namespace std;
+using namespace ngraph::snippets;
+
+class jit_snippet : public dnnl::impl::cpu::x64::jit_generator {
+public:
+    DECLARE_CPU_JIT_AUX_FUNCTIONS(jit_snippet)
+
+    ~jit_snippet() = default;
+
+    jit_snippet() : jit_generator() {
+    }
+
+    void generate() override {
+    }
+};
+
+#define CREATE_EMITTER(e_type) [this](const std::shared_ptr<ngraph::Node>& n) \
+    -> std::shared_ptr<ngraph::snippets::Emitter> {return std::make_shared<e_type>(h.get(), isa, n);};
+
+MKLDNNPlugin::CPUGenerator::CPUGenerator() : h(new jit_snippet()), isa(dnnl::impl::cpu::x64::avx2) {
+    // data movement
+    jitters[ngraph::opset1::Parameter().get_type_info()] = CREATE_EMITTER(NopEmitter);
+    jitters[ngraph::snippets::op::BlockedParameter().get_type_info()] = CREATE_EMITTER(NopEmitter);
+    jitters[ngraph::opset1::Result().get_type_info()] = CREATE_EMITTER(NopEmitter);
+    // jitters[ngraph::opset1::Constant().get_type_info()] = CREATE_EMITTER(); // Not supported
+
+    jitters[ngraph::snippets::op::Load().get_type_info()] = CREATE_EMITTER(LoadEmitter);
+    jitters[ngraph::snippets::op::VectorLoad().get_type_info()] = CREATE_EMITTER(LoadEmitter);
+    jitters[ngraph::snippets::op::ScalarLoad().get_type_info()] = CREATE_EMITTER(ScalarLoadEmitter);
+    jitters[ngraph::snippets::op::BroadcastLoad().get_type_info()] = CREATE_EMITTER(BroadcastLoadEmitter);
```

Comment:
why not `ngraph::snippets::op::BroadcastLoad::type_info`?

Reply:
I am an artist, this is how I see it 🤣 🤦